packages/yarnpkg-core/tests/Configuration.test.ts

```diff
@@ -0,0 +1,36 @@
+import {xfs, PortablePath}     from '@yarnpkg/fslib';
+import NpmPlugin               from '@yarnpkg/plugin-npm';
+
+import {Configuration, SECRET} from '../sources/Configuration';
+
+async function initializeConfiguration<T>(value: unknown, cb: (dir: PortablePath) => Promise<T>) {
+  return await xfs.mktempPromise(async dir => {
+    await Configuration.updateConfiguration(dir, value);
+
+    return await cb(dir);
+  });
+}
+
+describe(`Configuration`, () => {
+  it(`should hide secrets`, async () => {
+    await initializeConfiguration({
+      npmAuthToken: `my-token`,
+      npmScopes: {
+        myScope: {
+          npmAuthToken: `my-token`,
+        },
+      },
+    }, async dir => {
+      const configuration = await Configuration.find(dir, {
+        modules: new Map([[`@yarnpkg/plugin-npm`, NpmPlugin]]),
+        plugins: new Set([`@yarnpkg/plugin-npm`]),
+      });
+
+      const firstToken = configuration.getForDisplay(`npmAuthToken`);
+      const secondToken = configuration.getForDisplay(`npmScopes`).get(`myScope`).get(`npmAuthToken`);
```

Comment:
While keeping a unit test here is fine, I tend to prefer integration tests.

The reason for that is that because they are typically wider in scope, they also test behaviours closer from what real-life users will experience. For example, while you did test that `getForDisplay` works, you didn't test that it actually gets used by `yarn config get`. As a result, it's quite possible for someone to make an accidental regression during a refactoring.

Food for thought 🙂